modules/internal/ChapelArray.chpl

```diff
@@ -2923,11 +2923,63 @@ module ChapelArray {
     }
   }
 
+  // The following are the historical reshape() procedures that rely
+  // on copying the array's elements, proposed to be replaced by the
+  // pre-edition variants that follow
+
   /* Return a copy of the array ``A`` containing the same values but
      in the shape of the domain ``D``. The number of indices in the
      domain must equal the number of elements in the array. The
      elements of ``A`` are copied into the new array using the
-     default iteration orders over ``D`` and ``A``.  */
+     default iteration orders over ``D`` and ``A``.
+
+     .. note::
+
+        In addition to the above version of reshape(), there is
+        another experimental version that is available when compiling
+        with ``--edition=pre-edition`` that provides new features
+        including:
+
+        * creating a reshape() that aliases the original array rather
+          than copying it
+        * reshaping using a series of ranges rather than a domain
+        * reshaping to a 1D inferred-size array using an unbounded
+          range like ``1..``
+
+        Note that in order to capture an alias in a named variable, a
```

Comment:
Should this mention that not using `ref` will copy? E.g., "Note that in order to capture an alias in a named variable without copying the underlying array...".